test/cluster/test_data_resurrection_in_memtable.py

```diff
@@ -0,0 +1,141 @@
+#
+# Copyright (C) 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+#
+
+import json
+import logging
+import pytest
+import time
+
+from cassandra.cluster import ConsistencyLevel  # type: ignore
+from cassandra.query import SimpleStatement  # type: ignore
+
+from test.cluster.conftest import skip_mode
+from test.cluster.util import new_test_keyspace
+from test.pylib.manager_client import ManagerClient
+from test.pylib.util import wait_for_cql_and_get_hosts
+
+
+logger = logging.getLogger(__name__)
+
+
+async def run_test_cache_tombstone_gc(manager: ManagerClient, statement_pairs: list[tuple[str]]):
+    """Test for cache garbage collecting tombstones which cover data in the memtable.
+
+    1. Write a live row.
+    2. Write a tombstone to 2/3 replica (fail the write on node3 via error injection).
+    3. Run a repair so node3 also receives the tombstone.
+
+    At this stage, node1 and node2 have both the live row and the tombstone in
+    memtable, node3 has the live row in the memtable and the tombstone on disk.
+
+    4. Read the row from each node with CL=LOCAL_ONE. This will create an entry in cache
+       on node3, with the tombstone.
+       Check that population didn't drop the tombstone! #23291
+    5. Do another read round. This will use the existing entry in the cache.
+       Check that the cache read didn't drop the tombstone! #23252
+    """
+    cmdline = ["--hinted-handoff-enabled", "0", "--cache-hit-rate-read-balancing", "0", "--logger-log-level", "debug_error_injection=trace"]
+
+    node1 = await manager.server_add(cmdline=cmdline)
+    node2 = await manager.server_add(cmdline=cmdline)
+    node3 = await manager.server_add(cmdline=cmdline)
+
+    nodes = [node1, node2, node3]
```

Comment:
nit: can use `manager.servers_add` to add all 3 nodes at the same time.

However, to prepare for rf=#racks each node should be in a different rack so at the moment each one should be added independently given its own property_file

Reply:
I will change this to `servers_add()`, I also hope this will facilitate cluster reuse between the tests in this file.

We should have a `servers_add()` overload which takes a dict describing a topology, just like we did in `ccm.scylla_cluster.populate()`.

Reply:
Done in v2.

Reply:
> We should have a `servers_add()` overload which takes a dict describing a topology, just like we did in `ccm.scylla_cluster.populate()`.

YES. Do we have an issue tracking this?